tests/test_website.py

```diff
@@ -0,0 +1,115 @@
+import pytest
+import aiohttp.web
+import logging
+
+from collections import namedtuple
+from pyslackersweb.website import tasks
+
+SlackInviteTestParam = namedtuple("Param", "response data expected")
+
+
+@pytest.fixture
+async def website_client(app: aiohttp.web.Application, aiohttp_client, slack_client, redis):
+    app["redis"] = app["website_app"]["redis"] = redis
+
+    client = await aiohttp_client(app)
+
+    client.app["website_app"]["slack_client"] = slack_client
+    return client
+
+
+async def test_endpoint_index(website_client):
+    r = await website_client.get("/")
+
+    assert r.history[0].url.path == "/"
+    assert r.history[0].status == 302
+
+    assert r.status == 200
+    assert r.url.path == "/web"
+
+
+async def test_endpoint_slack(website_client):
+    r = await website_client.get("/web/slack")
+    assert r.status == 200
+
+
+@pytest.mark.parametrize(
+    "slack_client,data,expected",
+    (
+        SlackInviteTestParam(
+            response={},
+            data={"email": "error@example.com", "agree_tos": True},
+            expected="successAlert",
+        ),
+        SlackInviteTestParam(
+            response={}, data={"agree_tos": True}, expected="Missing data for required field"
+        ),
+        SlackInviteTestParam(
+            response={},
+            data={"email": "error@example.com", "agree_tos": False},
+            expected="There was an error processing your invite",
+        ),
+        SlackInviteTestParam(
+            response={},
+            data={"email": "foobar", "agree_tos": True},
+            expected="Not a valid email address",
+        ),
+        SlackInviteTestParam(
+            response={"body": {"ok": False, "error": "already_in_team"}},
+            data={"email": "error@example.com", "agree_tos": True},
+            expected="Reason: already_in_team",
+        ),
+        SlackInviteTestParam(
+            response={"body": {"ok": False, "error": "not_authed"}},
+            data={"email": "error@example.com", "agree_tos": True},
+            expected="Reason: not_authed",
+        ),
+        SlackInviteTestParam(
+            response={"status": 500},
+            data={"email": "error@example.com", "agree_tos": True},
+            expected="Reason: Error contacting slack API",
+        ),
+    ),
+    indirect=["slack_client"],
+)
+async def test_endpoint_slack_invite(website_client, data, expected):
+    r = await website_client.post(path="/web/slack", data=data)
+    html = await r.text()
+
+    assert r.status == 200
+    assert expected in html
+
+
+async def test_task_sync_github_repositories(caplog, redis):
+
+    async with aiohttp.ClientSession() as session:
+        result = await tasks.sync_github_repositories(session, redis)
+
+    assert result
+
+    for record in caplog.records:
```

Comment:
I really do not want to test logging implementation, that seems brittle and not like something to ensure doesn't change.

Reply:
The goal is not to check the logging implementation but to ensure not exceptions where raised. It kinds of loop back to the `return` / `raise` issue of the tasks.

IMO it's fine for now.

Later on, we should split those tasks into two functions. One handling the logging / exception catching for apscheduler and a second doing the actual work that we can test